port/unix/omrsysinfo.c

```diff
@@ -6474,12 +6592,23 @@ omrsysinfo_cgroup_subsystem_iterator_init(struct OMRPortLibrary *portLibrary, ui
 	Assert_PRT_true(NULL != state);
 	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
 #if defined(LINUX) && !defined(OMRZTPF)
+	size_t omrCgroupMemoryMetricMapSize = 0;
 	state->count = 0;
 	state->subsystemid = subsystem;
 	state->fileMetricCounter = 0;
+	if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+		omrCgroupMemoryMetricMap = omrCgroupMemoryMetricMapV1;
+		omrCgroupMemoryMetricMapSize = OMR_CGROUP_MEMORY_METRIC_MAP_V1_SIZE;
+	} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE)) {
+		omrCgroupMemoryMetricMap = omrCgroupMemoryMetricMapV2;
+		omrCgroupMemoryMetricMapSize = OMR_CGROUP_MEMORY_METRIC_MAP_V2_SIZE;
+	} else {
+		Trc_PRT_Assert_ShouldNeverHappen();
+	}
+
```

Comment:
Could we store omrCgroupMemoryMetricMap into the state? That way we can store the needed struct once at init instead of in both iterator_metricKey and iterator_next (every init/metrickey/next iteration only applies to one subsystem). Then we could do at init
```
if v1:
  switch subsystem
    memoryv1
    cpuv1
    ...
if v2:
  switch subsystem
    memoryv2
    ...
```

Reply:
Exposing it as `OMRCgroupSubsystemMetricMap` in `OMRCgroupMetricIteratorState` won't work. `OMRCgroupSubsystemMetricMap` is only defined in `omrsysinfo.c` whereas `OMRCgroupMetricIteratorState` is defined in `omrport.h`. We can store it as `void *` while masking the actual data type. But, this will expose that struct to the user. If a user writes bad memory to it or has other evil plans, it will impair the API and lead to security issues.

Reply:
Added function to get OMRCgroupSubsystemMetricMap (cpu and cpuset still pending)
Not used in iterator_init since it wants the size instead of the map